saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/apply/DefaultEnrichmentTagApplierTest.java

```diff
@@ -0,0 +1,160 @@
+    
+    @Mock private JudgmentUpdater<Object> judgmentUpdater;
+    
+    @Mock private Judgment judgment;
+    
+    @Mock private EnrichmentTag enrichmentTag;
+ 
+    
+    private DefaultEnrichmentTagApplier<String, Object> defaultEnrichmentTagApplier = new DefaultEnrichmentTagApplier<>(HANDLED_TAG_TYPE);
+    
+    
+    
+    @Before
+    public void before() {
+        
+        initMocks(this);
+        
+        defaultEnrichmentTagApplier.setJsonStringParser(jsonStringParser);
+        defaultEnrichmentTagApplier.setEnrichmentTagValueConverter(enrichmentTagValueConverter);
+        defaultEnrichmentTagApplier.setJudgmentUpdater(judgmentUpdater);
+        
+        when(enrichmentTag.getTagType()).thenReturn(HANDLED_TAG_TYPE);
+        
+    }
+    
+    
+    //------------------------ TESTS --------------------------
+    
+    
+    @Test
+    public void handlesEnrichmentTagType_DoesNotHandle() {
+        
+        // execute & assert
+        assertFalse(defaultEnrichmentTagApplier.handlesEnrichmentTagType("XXX"));
+        
+        // assert
+        verifyZeroInteractions(enrichmentTagValueConverter, jsonStringParser, judgmentUpdater);
+
+    }
+    
+    @Test
+    public void handlesEnrichmentTagType_Handles() {
+        
+        // execute & assert
+        assertTrue(defaultEnrichmentTagApplier.handlesEnrichmentTagType(HANDLED_TAG_TYPE));
+        
+        // assert
+        verifyZeroInteractions(enrichmentTagValueConverter, jsonStringParser, judgmentUpdater);
+
+    }
+    
+    
+    @Test(expected=IllegalArgumentException.class)
+    public void applyEnrichmentTag_NotHandledTagType() {
+        
+        // given
+        when(enrichmentTag.getTagType()).thenReturn("ABC");
+        
+        // execute
+        defaultEnrichmentTagApplier.applyEnrichmentTag(judgment, enrichmentTag);
+        
+        // assert
+        verifyZeroInteractions(enrichmentTagValueConverter, jsonStringParser, judgmentUpdater);
+
+    }
+    
+    
+    @Test(expected=NullPointerException.class)
+    public void applyEnrichmentTag_NullJudgment() {
+        
+        // execute
+        defaultEnrichmentTagApplier.applyEnrichmentTag(null, enrichmentTag);
+        
+        // assert
+        verifyZeroInteractions(enrichmentTagValueConverter, jsonStringParser, judgmentUpdater);
+
+    }
+    
+    
+    @Test(expected=NullPointerException.class)
+    public void applyEnrichmentTag_NullEnrichmentTag() {
+        
+        // execute
+        defaultEnrichmentTagApplier.applyEnrichmentTag(judgment, null);
+        
+        // assert
+        verifyZeroInteractions(enrichmentTagValueConverter, jsonStringParser, judgmentUpdater);
+    }
+    
+    
+    @Test(expected=IllegalArgumentException.class)
+    public void applyEnrichmentTag_JsonParseEx() throws Exception {
+        
+        // given
+        when(jsonStringParser.parseAndValidate(Mockito.anyString())).thenThrow(new IllegalArgumentException());
```

Comment:
It is because of JsonParseException but in such cases I don't care and just throw Exception, because in my view it doesn't really matter.